src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.health = health;
+    this.name = name;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(object) {
+    if (object instanceof Herbivore && !object.hidden) {
+      object.health -= 50;
+    }
+
+    if (object.health <= 0) {
+      const deadAnimal = Animal.alive.findIndex((victim) => victim.health <= 0);
+
+      Animal.alive.splice(deadAnimal, 1);
```

Comment:
The logic for removing a dead animal from the `Animal.alive` array is incorrect. The `findIndex` method is used to find the index of the first animal with health less than or equal to 0, but this could lead to incorrect behavior if multiple animals have health below 0. Consider using a more robust method to ensure only the specific `object` that was bitten and has health <= 0 is removed from the `alive` array.